querybook/webapp/components/DataDocGitHub/GitHubDirectory.tsx

```diff
@@ -0,0 +1,163 @@
+import { Form, Formik } from 'formik';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import * as Yup from 'yup';
+
+import { GitHubResource } from 'resource/github';
+import { AsyncButton } from 'ui/AsyncButton/AsyncButton';
+import { FormWrapper } from 'ui/Form/FormWrapper';
+import { SimpleField } from 'ui/FormikField/SimpleField';
+import { Icon } from 'ui/Icon/Icon';
+import { Loading } from 'ui/Loading/Loading';
+import { Message } from 'ui/Message/Message';
+
+interface IProps {
+    docId: number;
+    linkedDirectory?: string | null;
+    onLinkDirectory: (directory: string) => Promise<void>;
+}
+
+const validationSchema = Yup.object().shape({
+    directory: Yup.string().notRequired(),
+});
+
+export const GitHubDirectory: React.FC<IProps> = ({
+    docId,
+    linkedDirectory,
+    onLinkDirectory,
+}) => {
+    const [directories, setDirectories] = useState<string[]>([]);
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
+    const [isLoading, setIsLoading] = useState<boolean>(false);
+
+    const fetchDirectories = useCallback(async () => {
+        setIsLoading(true);
+        try {
+            const { data } = await GitHubResource.getDirectories(docId);
+            setDirectories(data.directories);
+        } catch (error) {
+            console.error('Failed to fetch directories:', error);
+            setErrorMessage('Failed to fetch directories. Please try again.');
+        } finally {
+            setIsLoading(false);
+        }
+    }, [docId]);
+
+    useEffect(() => {
+        fetchDirectories();
+    }, [fetchDirectories]);
+
+    const directoryOptions = useMemo(
+        () =>
+            directories.map((dir) => ({
+                label: dir,
+                value: dir,
+            })),
+        [directories]
+    );
+
+    const handleSubmit = async (values: { directory: string }) => {
+        const directory = values.directory || 'datadocs';
```

Comment:
have a const for the default directory "datadocs" and make sure it is in sync with the backend if it is also used in backend.